app/api/csv.py

```diff
@@ -125,20 +180,331 @@ def row(r):
+
+        return answers
+
+    return convert_to_csv(seller_responses, row)
+
+
+def generate_specialist_opportunities_csv(specialistOpportunities):
+    # converts a brief response into an ordered dict
+    def row(r):
+        answers = od()
+        answers.update({'opportunity_id': r['id']})
+        answers.update({'internal_reference': r['internalReference']})
+        answers.update({'title': r['title']})
+        answers.update({'buying_agency': r['organisation']})
+        answers.update({'raised_by': r['email_address']})
+        answers.update({'role_responsibilities': r['summary']})
+        answers.update({'location': ', '.join(r['location'])})
+        answers.update({'service': r['areaOfExpertise']})
+        answers.update({'open_to': r['openTo']})
+        answers.update({'number_of_sellers_approached': r['numberOfSuppliers']})
+        answers.update({'sellers': format_sellers(r['sellers'])})
+        answers.update({'essential_criteria': format_criteria(r.get('essentialRequirements') or '')})
+        answers.update({'desirable_criteria': format_criteria(r.get('niceToHaveRequirements') or '')})
+        answers.update({'evaluation_steps': ', '.join(r['evaluationType'])})
+        answers.update({'rate_format': r['preferredFormatForRates']})
+        answers.update({'max_rate_cap': r['maxRate']})
+        answers.update({'security_clearance_requirement': r['securityClearance']})
+        answers.update({'start_date': r['startDate']})
+        answers.update({'contract_length': r['contractLength']})
+        answers.update({'contract_extension': r['contractExtensions']})
+        answers.update({'awarded_to': r['awarded_to']})
+        answers.update({'time_draft_created': r['created_at']})
+        answers.update({'time_published': r['published_at']})
+        answers.update({'time_closed': r['closed_at']})
+        answers.update({'time_awarded': r['time_awarded']})
+
+        for k, v in answers.items():
+            answers[k] = csv_cell_sanitize(v)
+
+        return answers
+
+    return convert_to_csv(specialistOpportunities, row)
+
+
+def generate_atm_opportunities_csv(atmOpportunities):
+    # converts a brief response into an ordered dict
+    def row(r):
+        answers = od()
+        answers.update({'opportunity_id': r['id']})
+        answers.update({'title': r['title']})
+        answers.update({'raised_by': r['email_address']})
+        answers.update({'open_to': r['openTo']})
+        answers.update({'category_open_to': r['areaOfExpertise']})
+        answers.update({'buying_agency': r['organisation']})
+        answers.update({'summary': r['summary']})
+        answers.update({'state': ', '.join(r['location'])})
+        answers.update({'evaluation_steps': ', '.join(r['evaluationType'])})
+        answers.update({'commence_date': r['startDate']})
+        eval_crit_list = r.get('evaluationCriteria', [])
+        eval_crit_inner_dict = {}
+        eval_crit_outer_dict = {}
+        outerIndex = 1
```

Comment:
These `eval_crit` and `outerIndex` variables don't appear to be used. Is something missing?

Reply:
removed